### PR TITLE
fix(nvidia): use NVML + lspci to detect NVIDIA GPUs (without running nvidia-smi)

### PR DESCRIPTION
**Pull request by gyuho**

### CODE REVIEW

File: components/accelerator/nvidia/query/detect.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
+)
+
+// Returns true if the local machine has NVIDIA GPUs installed.
+func GPUsInstalled(ctx context.Context) (bool, error) {
+	smiInstalled := SMIExists()
+	if !smiInstalled {
+		return false, nil
+	}
+	log.Logger.Info("nvidia-smi installed")
```
Comment (xiang90) [inline components/accelerator/nvidia/query/detect.go:24]: should be debug level?

File: components/accelerator/nvidia/query/detect.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+	}
+	log.Logger.Infow("nvidia PCI devices found", "devices", len(pciDevices))
+
+	// now that we have the NVIDIA PCI devices,
+	// call NVML C-based API for NVML API
+	productName, err := LoadProductName(ctx)
+	if err != nil {
+		return false, err
+	}
+	log.Logger.Infow("detected nvidia gpu", "product", productName)
```
Comment (xiang90) [inline components/accelerator/nvidia/query/detect.go:43]: productName can also be a network card. do we check the type of the device?

Comment (gyuho) [reply components/accelerator/nvidia/query/detect.go]: It's calling the device.GetName, so changed this logging to GPU device name.

File: components/accelerator/nvidia/query/detect.go (hunk 1 above)
Comment (gyuho) [reply components/accelerator/nvidia/query/detect.go]: Done

File: components/accelerator/nvidia/query/detect.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+func GPUsInstalled(ctx context.Context) (bool, error) {
+	smiInstalled := SMIExists()
+	if !smiInstalled {
+		return false, nil
+	}
+	log.Logger.Debugw("nvidia-smi installed")
+
+	// now that nvidia-smi installed,
+	// check the NVIDIA GPU presence via PCI bus
+	pciDevices, err := ListPCIs(ctx)
```
Comment (xiang90) [inline components/accelerator/nvidia/query/detect.go:28]: it is only useful for printing?

Comment (gyuho) [reply components/accelerator/nvidia/query/detect.go]: Renamed to `ListNVIDIAPCIs`. It's only listing the devices with NVIDIA name in it, thus checking whether the host has NVIDIA devices.

**Review by cardyok — APPROVED** [review]

**Pull request merged**